pkg/schemadsl/compiler/translator.go

```diff
@@ -453,9 +453,7 @@ func translateExpressionDirect(tctx *translationContext, expressionNode *dslNode
 		if err != nil {
 			return nil, err
 		}
-		leftOps := collapseOps(leftOperation, lookup)
-		rightOps := collapseOps(rightOperation, lookup)
-		ops := append(leftOps, rightOps...)
+		ops := append(collapseOps(leftOperation, lookup), collapseOps(rightOperation, lookup)...)
```

Comment:
Please leave it as it was, this is a cosmetic change and it seemed easier to parse visually before

Reply:
I agree that it was easier to parse visually - technically this prevents a [possible memory issue](https://stackoverflow.com/questions/78685528/what-is-the-idiomatic-method-and-style-of-combining-two-slices-in-go) stemming from this lint finding: `appendAssign: append result not assigned to the same slice (gocritic)`

When we do append(leftOps, rightOps...), if leftOps isn't large enough, there would be a new underlying array created. Since we're assigning it back to ops and not leftOps, The original leftOps slice header would still point to the old memory. The linter is upset at that potential reallocation. At least I think that's what's going on here.

I don't see this being a huge issue considering the small memory this is using anyways. I'll go ahead and revert it.